core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/compression/GzipCompressor.java

```diff
@@ -38,22 +38,39 @@ public String compressorType() {
     }
 
     @Override
-    public InputStream compress(InputStream inputStream) {
+    public byte[] compress(byte[] content) {
         try {
-            byte[] content = IoUtils.toByteArray(inputStream);
             ByteArrayOutputStream compressedOutputStream = new ByteArrayOutputStream();
             GZIPOutputStream gzipOutputStream = new GZIPOutputStream(compressedOutputStream);
             gzipOutputStream.write(content);
             gzipOutputStream.close();
+            return compressedOutputStream.toByteArray();
+        } catch (IOException e) {
+            throw SdkClientException.create(e.getMessage(), e);
```

Comment:
Can we throw ` new UncheckedIOException(e);` , I see in other places we have used this too.

Reply:
+1